client/src/lib/solana/instructions/instruction.ts

```diff
@@ -1,10 +1,52 @@
 import {
-  Enum,
-  Assignable,
   add_enum_to_schema,
   add_struct_to_schema,
+  Assignable,
+  AssignableBoolean,
+  AssignableI64,
+  Enum,
 } from '../solanaBorsh';
 import { InstructionData } from '../model/InstructionData';
+import { AssignablePublicKey } from '../model/AssignablePublicKey';
+import { PublicKey } from '@solana/web3.js';
+import { TransactionState } from '../model/TransactionState';
+import { AccountOperation } from '../model/AccountOperation';
+import { InstructionOperation } from '../model/InstructionOperation';
+import { ProposeTransactionSigners } from '../model/ProposeTransactionSigners';
+
+export class ProposeTransaction extends Assignable<ProposeTransaction> {
```

Comment:
Is there a "solana-generator" way to do this, that is better than the borsh way? i.e. does not need the weird props constructor that breaks type safety?

Reply:
I got a type error with this and got annoyed so fixed it in the later PR :). The fix is to require by the type system all the properties in props